src/Core/Grid/Presenter/GridPresenter.php

```diff
@@ -0,0 +1,217 @@
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Grid\Presenter;
+
+use PrestaShop\PrestaShop\Core\Grid\Action\GridActionInterface;
+use PrestaShop\PrestaShop\Core\Grid\Column\ColumnCollectionInterface;
+use PrestaShop\PrestaShop\Core\Grid\Column\ColumnInterface;
+use PrestaShop\PrestaShop\Core\Grid\Exception\MissingColumnInRowException;
+use PrestaShop\PrestaShop\Core\Grid\GridInterface;
+
+/**
+ * Class GridPresenter is responsible for presenting grid
+ */
+final class GridPresenter implements GridPresenterInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function present(GridInterface $grid)
+    {
+        $gridArray = [
+            'id' => $grid->getDefinition()->getId(),
+            'name' => $grid->getDefinition()->getName(),
+            'filter_form' => $grid->getFilterForm()->createView(),
+        ];
+
+        $gridArray['columns'] = $this->presentColumns($grid);
+        $gridArray['bulk_actions'] = $this->presentBulkActions($grid);
+        $gridArray['actions'] = $this->presentGridActions($grid);
+
+        $gridArray['data'] = [
+            'rows' => $this->presentRows($grid),
+            'rows_total' => $grid->getData()->getRowsTotal(),
+            'query' => $grid->getData()->getQuery(),
+        ];
+
+        $gridArray['pagination'] = [
+            'offset' => $grid->getSearchCriteria()->getOffset(),
+            'limit' => $grid->getSearchCriteria()->getLimit(),
+        ];
+
+        $gridArray['sorting'] = [
+            'order_by' => $grid->getSearchCriteria()->getOrderBy(),
+            'order_way' => $grid->getSearchCriteria()->getOrderWay(),
+        ];
+
+        return $gridArray;
+    }
+
+    /**
+     * Present grid columns
+     *
+     * @param GridInterface $grid
+     *
+     * @return array Presented columns
+     */
+    private function presentColumns(GridInterface $grid)
+    {
+        $columnsArray = [];
+
+        $columns = $grid->getDefinition()->getColumns();
+        $positions = [];
+
+        /** @var ColumnInterface $column */
+        foreach ($columns as $key => $column) {
```

Comment:
as it is a ColumCollection, can you move this business logic into ColumCollection class please?

Reply:
should collection be responsible for it's presentation? if yes, i can move this to `ColumnCollection` and add method `present()`

Reply:
it's twig who is responsible for presentation :D, the collection is responsible for everything linked to itself ;)